File: bin/petozwell.js
#!/usr/bin/env node
import { readFile } from 'node:fs/promises';
import { parseArgs } from 'node:util';
import { installPet, resolveCodexHome } from '../lib/install.js';

const help = `Usage: petozwell [install] [options]

Install Ozwell into your Codex pets folder.

Options:
  --codex-home <path>  Override CODEX_HOME (default: ~/.codex)
  --dry-run            Show the destination without changing files
  --force              Back up an existing Ozwell folder before replacing it
  --help, -h           Show this help
  --version, -v        Show the installer version

After installing, open Settings > Pets > Refresh and select Ozwell.
Use Wake Pet or /pet to show him. Installation does not change app settings.
`;

try {
  const { values, positionals } = parseArgs({
    options: {
      'codex-home': { type: 'string' },
      'dry-run': { type: 'boolean', default: false },
      force: { type: 'boolean', default: false },
      help: { type: 'boolean', short: 'h' },
      version: { type: 'boolean', short: 'v' },
    },
    allowPositionals: true,
  });
  if (values.help) {
    console.log(help);
  } else if (values.version) {
    const pkg = JSON.parse(await readFile(new URL('../package.json', import.meta.url), 'utf8'));
    console.log(pkg.version);
  } else {
    if (positionals.length > 1 || (positionals.length === 1 && positionals[0] !== 'install')) {
      throw new Error('Unknown command. Run petozwell --help for usage.');
    }
    const result = await installPet({
      codexHome: resolveCodexHome(values['codex-home']),
      force: values.force,
      dryRun: values['dry-run'],
    });
    if (result.status === 'dry-run') {
      console.log(`Would install Ozwell into ${result.destination}`);
      if (result.conflict) console.log('An existing installation differs. Use --force to back it up and replace it.');
    } else {
      console.log(`${result.status === 'unchanged' ? 'Ozwell is already installed at' : 'Installed Ozwell into'} ${result.destination}`);
      if (result.backup) console.log(`Previous installation saved at ${result.backup}`);
      console.log('Open Settings > Pets > Refresh, select Ozwell, then choose Wake Pet if needed.');
    }
  }
} catch (error) {
  console.error(`Ozwell: ${error.message}`);
  process.exitCode = 1;
}

File: Formula/petozwell.rb
class Petozwell < Formula
  desc "Install Ozwell, the yellow octopus pet for Codex desktop"
  homepage "https://github.com/mieweb/petozwell"
  url "https://github.com/mieweb/petozwell/releases/download/v1.0.0/petozwell-1.0.0.tgz"
  sha256 "b88f1385930f4c786bb3f3fd6b05794a685aa9cde7b1be6c6dc77810e78e70e2"
  license :cannot_represent

  depends_on "node"

  def install
    libexec.install "bin", "lib", "ozwell", "package.json"
    (bin/"petozwell").write <<~SH
      #!/bin/sh
      exec "#{formula_opt_bin("node")}/node" "#{libexec}/bin/petozwell.js" "$@"
    SH
  end

  def caveats
    <<~EOS
      Run petozwell to install Ozwell into your Codex pets folder.
      Then open Settings > Pets > Refresh and select Ozwell.
      The installer honors CODEX_HOME, or accepts --codex-home PATH.
      To replace an existing version while keeping a backup, run:
        petozwell --force
    EOS
  end

  test do
    codex_home = testpath/"codex home"
    system bin/"petozwell", "--codex-home", codex_home
    manifest = codex_home/"pets/ozwell/pet.json"
    assert_path_exists manifest
    assert_equal "Ozwell", JSON.parse(manifest.read).fetch("displayName")
    assert_path_exists codex_home/"pets/ozwell/spritesheet.png"
    assert_match "already installed", shell_output("#{bin}/petozwell --codex-home '#{codex_home}'")
  end
end
